db/collection.go

```diff
@@ -358,11 +358,15 @@ func (t *Txn) Commit() error {
 	if err != nil {
 		return err
 	}
-	if err := t.collection.db.dispatcher.Dispatch(events); err != nil {
-		return err
+	if len(events) > 0 {
+		if err := t.collection.db.dispatcher.Dispatch(events); err != nil {
+			return err
+		}
 	}
-	if err := t.collection.db.notifyTxnEvents(node); err != nil {
-		return err
+	if node != nil {
+		if err := t.collection.db.notifyTxnEvents(node); err != nil {
+			return err
+		}
```

Comment:
Good points. I should have made a note about this change. I added these conditionals because there were some panics on the tests... the `mockEventCodec` returns a `nil` node:

```
func (dec *mockEventCodec) Create([]core.Action) ([]core.Event, format.Node, error) {
	dec.called = true
	return nil, nil, nil
}
```

This was causing a panic downstream on the `localEventBus`: 

```
func (leb *localEventsBus) Listen() *LocalEventListener {
	l := &LocalEventListener{
		listener: leb.bus.Listen(),
		c:        make(chan format.Node),
	}

	go func() {
		for v := range l.listener.Channel() {
			events := v.(format.Node)
			l.c <- events
		}
		close(l.c)
	}()

	return l
}
```

Specifically, `events := v.(format.Node)` is not safe when `v` is nil. 

In any case, I can make your change... having both events and node empty is still not valid.